pkg/vm/vm.go

```diff
@@ -503,20 +503,24 @@ func (v *VM) execute(ctx *Context, op Instruction) {
 		v.estack.PushVal(new(big.Int).Mod(a, b))
 
 	case SHL:
-		b := v.estack.Pop().BigInt()
-		if b.Int64() == 0 {
+		b := v.estack.Pop().BigInt().Int64()
+		if b == 0 {
 			return
+		} else if b < -256 || b > 256 {
+			panic("SHL: operand must be between -256 and 256")
 		}
 		a := v.estack.Pop().BigInt()
-		v.estack.PushVal(new(big.Int).Lsh(a, uint(b.Int64())))
+		v.estack.PushVal(new(big.Int).Lsh(a, uint(b)))
 
 	case SHR:
-		b := v.estack.Pop().BigInt()
-		if b.Int64() == 0 {
+		b := v.estack.Pop().BigInt().Int64()
+		if b == 0 {
 			return
+		} else if b < -256 || b > 256 {
+			panic("SHR: operand must be between -256 and 256")
```

Comment:
BTW, maybe it's better to unify these opcode implementations as they only differ in the last call. And this line can easily live without `SHR/SHL` prefix as it's printed anyway during `recover()`.

Reply:
Yeah, I have unified them.